### PR TITLE
Password page

### LINKED ISSUE #43: Password page core functionality
# Overview
Implement the core functionality of the password page. 

## Frontend (UI/UX)
- text box and button to submit password
- checkbox to remember password
- validate the password against some criteria (the criteria for a strong password is up to you, just make sure it's not too strict to annoy the user)
- style and design is TBD so don't worry about it too much (apply some styling still for it to look decent)

## Backend (CRUD operations)
- Think about how we can store the password if the user decided to check the remember me textbox

## Acceptance Criteria
- [ ] Basic frontend components
- [ ] Validating the password and showing an error if the password is not "strong"
- [ ] Basic styling (in `styles/password-page.css`)
- [ ] A method to store the password if the user checks the "remember me" checkbox
- [ ] Proper documentation of the functions
- [ ] Unit testing and E2E testing if applicable

### PR DESCRIPTION
**Pull request by matt0923**
## Tracking Info

Resolves issue #43

## Changes

We added functionality to the password page part of the website, including:
. A main page for entering the password
. A create password page
. A create pin page
. A help page
. A "remember me" option. 

## Testing

No testing yet, just visually made sure that the feature covered all aspects of the issue and worked as it was intended to (testing will come in the next few days). 

## Confirmation of Change

image

### CODE REVIEW

**Review by lindseyrapp — DISMISSED** [review]
This is great progress, thanks so much guys. It's looking great, the styling is awesome. A couple things to keep in mind, we want to make sure that we are redirected back to the password page after creating a new password & pin. Additionally, we want the password/pin fields to contain the new created password/pin after entered. So far, so good, these are just some next steps. 
Partially resolves #63 (still needs some changes but great progress!)

**Comment by lindseyrapp** [comment]
It looks like there are some issues in your code that are not passing the Lint check. I resolved the one in preload.js but there are others in main & some other files you may want to check out.

File: source/main.js [hunk 1]
```diff
@@ -244,3 +244,19 @@ ipcMain.handle('update-markdown-entry', async (event, arg) => {
         fs.writeFileSync(entries[entryIndex].fileName, arg.markdownContent, 'utf-8');
     }
 });
+
```
Comment (RamtinTJB) [inline source/main.js]: Make sure to add proper JSDoc documentation to these methods. Refer to the ones right above this to see examples

File: source/password.html [hunk 2]
```diff
@@ -1,13 +1,66 @@
...
+    <div class="background-layer1">
+        <div class="background-layer2">
+            <div class="background-layer3">
+                <div class="logoBox">
+                    <img src="images/infiniteLogo.jpeg" alt="team logo">
+                    <p>Infinite Loops</p>
+                </div>
+                <div class="helpBox">
+                    <img src="images/questionMark.png" alt="helpBox" class="clickable-image" id="triggerPopup">
+                    <div id="myModal" class="modal">
```
Comment (RamtinTJB) [inline source/password.html:20]: Thanks for adding a help popup! Just make sure that the background is opaque and the text is readable on different screen sizes (at least different desktop sizes, don't worry about readability on mobile and tablet for now)

File: source/scripts/createPassword.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+function init() {
+    const button = document.querySelector('button');
+    const input = document.getElementById('password');
+    const alert = document.getElementById('strongPassword');
+    const passwordPage = document.getElementById('passwordPage');
+    const hint = document.getElementById('hint');
+    let password;
+    let pin;
+    button.addEventListener('click', () => {
+        if (input.value.length < 6) {
```
Comment (RamtinTJB) [inline source/scripts/createPassword.js:10]: Does this mean that the pin can only be less than 6 characters long? Do you guys think we can have a variable that keeps track of whether the user is entering the pin or the password and then check that variable in this if statement instead?

**Review by RamtinTJB — CHANGES_REQUESTED** [review]
Great job on this guys! I just left a few comments once these are resolved we can go ahead and merge. Also please make sure to add JSDoc documentation to all the functions and any piece of code that handles logic.

**Review by RamtinTJB — APPROVED** [review]
Great work guys! Thank you for writing the documentation and addressing the comments!

**Pull request merged**